### DISCUSSION TITLE
Can't Install Windows 11 ARM 22579 on Apple M1. No Internet. Can't Skip that.

### DISCUSSION THREAD

**Discussion by Rembrandt001** [utmapp/UTM#3795] [UTM Help (macOS)]
Can't get a Internet-connection at the install process. There is it only shows next grey an so I can't skip that. Any solution?

**Comment by Codex-** [comment]
Experiencing the same.

Following: https://mac.getutm.app/gallery/windows-11-arm
Using image: Windows11_InsiderPreview_Client_ARM64_en-us_22579.VHDX

Results in:
![image](https://user-images.githubusercontent.com/8297838/159409285-2d9229f3-c14c-461f-a8ec-5b53f0cd82b8.png)

I've tried:
- disabling the network completely
- setting manual addresses
- forwarding

Glad to see it's not just me. Maybe it works with an old image, haven't found one that works yet though

**Reply by MSamiulHasnat** [reply]
I'm also having the same issue. Did you fix it?

**Reply by tony1tf** [reply]
There's lots of info here - it's also on the UTM help pages:
https://pureinfotech.com/bypass-internet-connection-install-windows-11/
I was having problems with bypassing the network stage (at step 19), until I realised that the command required is OOBE/BYPASSNRO is all capital letters - not '0' since the 'O' is for "OUT OF BOX EXPERIENCE"!!

**Reply by chrisherman** [reply]
just to add:
a) capitalization is not required
b) if you cannot locate the backslash (on my kb next to left shift key), you can do a "cd oobe" and "bypassnro". Each followed by pressing enter-key

**Reply by gitblanc** [reply]
it also happens to me :/

**Reply by bzh32** [reply]
Choose Network Mode: Shared Network
Emulated Network Card: virtue-net-pci
Settings>Network. V4.09 of UTM

**Comment by guibernardi7** [comment]
Hello Guys, I was stuck in this same screen so I found a solution from the video below, the guy install win 10 but it works for win 11 I tried my self,
please look the video in minute 8:35
https://www.youtube.com/watch?v=KsNoGPczavw

**Reply by Rembrandt001** [reply]
He find a way. Download a 11 ISO any. Unzip it. Go into sources. Open appraiserres.dll  in an Editor. Search for TPM. Del all things 1 or to lines) with TPM. I Guess 20 to 25 entries. Save it. Make a new ISO an Install Win 11.

**Reply by GabrielPyva** [reply]
man that video solved perfectly on my machine. Thank you!

**Comment by tallLeRoy** [comment]
I was able to get it installed on my M1 Mac mini.

Here is the solution.

Create the VM using the ARM64 VHDX downloaded from Microsoft.

Start the VM and immediately press Escape, use the UEFI bios to change the resolution to 1024x768. F10 ( On my Apple keyboard, I needed to press the fn key with F10. )to save then ESC twice and Restart. 

Let windows boot to the first screen. While it is doing its thing, use the CD icon on the top of the window to attach the spice-guest-tools-xxxx.iso to the VM.

When Windows asks you for the Country. Stop right there. Press Shift-F10 ( On my Apple Keyboard, I needed to press the fn key with Shift and F10. ) to bring up a Windows command line.
CD to D:
then run the spice-guest-tools-xxxxx.exe
reboot after the install.

You should be able to get past that screen for a full install.

The TPM and Secure boot will still keep you from updating the release, but it works ok.

LeRoy

**Reply by Codex-** [reply]
This was perfect, thank you so much! 🎉

**Reply by tallLeRoy** [reply]
Glad you found it helpful. I don't want to know how many failures I had coming up with that solution over two days. Lucky that the hang happened early in the install cycle.

**Reply by kesiganes** [reply]
thank you, you're lifesaver

**Reply by atlcell** [reply]
thank you sir

**Reply by fauxlimey** [reply]
Thanks for this. So cool. I had an issue though... I changed display resolution successfully, then I used the Drive Image Options button to change to spice-guest-tools-0.164.3.iso but when I enter CD to D: in the command line I get "The system cannot find the path specified." Then I found that you can go all the way to the internet screen and press Shift + Control + F3 (I have no Fn key on my keyboard). That reboots you into an admin page, from which you can run splice-guest-tools.exe. Then reboot and finish the setup.

**Reply by lbrdev** [reply]
[fauxlimey](https://github.com/fauxlimey) try to use not `cd D:\` just `D:\`
Then `dir` shows .exe file.

**Reply by ajax-finch** [reply]
@lbrdev or just `start d:\spice-guest-tools-xxxxx.exe`

**Reply by mandakhsw** [reply]
huge thanks!!!

**Reply by GetOutOfThatGarden** [reply]
Hey, this worked for me too. Thanks a lot!

**Reply by jimmychiang-sb** [reply]
It works! Many thanks!

**Reply by caroga1073** [reply]
Muchas Gracias :)

**Reply by FacundoVelazco** [reply]
Crack master de la vida, semental, animal, terremoto, maremoto, bestia, 10 de 10. Thanks, greets from argentina

**Reply by maxd-out83** [reply]
Thank you so much!!!

**Reply by conornee07** [reply]
How do you get to UEFI bios

**Reply by LukasBombach** [reply]
Thanks this worked for me.

For the others: The UEFI bios step is optional and when you do the spice-tools-thing you get a proper screen resolution anyhow.

**Reply by renannprado** [reply]
In case you're wondering where the resolution option is, after you land on the bios screen:

1. Go to **Device Manager**

![image](https://github.com/utmapp/UTM/assets/2359543/57447ebe-4b5d-4e6d-9e8f-fa949b908e5d)

2. Then **OVMF Platform Configuration**

![image](https://github.com/utmapp/UTM/assets/2359543/c5c747ba-c459-4cd5-9036-b1231246b279)

3. Finally change the screen resolution

![image](https://github.com/utmapp/UTM/assets/2359543/5e87c812-1ba1-4866-857f-1b57f4363caa)

**Comment by ioapan** [comment]
Hi, I'm stuck at "press the fn key with F10". Which is the fn key on the Mac keyboard? Anything I press with F10 mutes the sound. Any ideas?

Update: I've found it:
Click the Apple menu in the top-left corner of the screen, then select System Preferences.
Click the Keyboard icon, then check the box marked “Use all F1, F2, etc. keys as standard function keys.”

**Reply by tallLeRoy** [reply]
For my wide keyboard, fn is just below the F13 key. For the narrow keyboard, fn is on the lower left corner.

**Reply by ioapan** [reply]
I can't see this my keyboard: https://www.apple.com/shop/product/MMMR3LL/A/magic-keyboard-with-touch-id-and-numeric-keypad-for-mac-models-with-apple-silicon-us-english-black-keys?fnode=b0ca97e089119db89faec2ce8160e3a75e2f9ca6b703fd9b2c6a1b058914c3969383224783fde7655ce3daf9b577401ea06c4472c873fef74449c0e204a05f17c4ec0119d74510dcd19ddb83ad32afcb308a3294d7a09c596812da8831f85dcad52250309d6b16b3340c982d494a3058&fs=fh%3D40a2%252B8a4b3a

**Reply by Zenexer** [reply]
@ioapan It's right below your F13 key.  It has a globe icon on it.

**Comment by DimitarNestorov** [comment]
What worked for me (also no need for a Microsoft account):

1. Edit the VM and set Network Mode to None
2. Boot up the VM, click <kbd>Shift</kbd> + <kbd>F10</kbd>, type `oobe\bypassnro` in the Command Prompt, and hit <kbd>Enter</kbd>
4. Wait for the VM to reboot and click "I don't have internet"

After I got to the desktop I installed SPICE (mounted as a CD) and reverted the Network Mode setting and was good to go.

**Reply by calebmogbolu** [reply]
This worked for me too, thanks

**Reply by tonyaxo** [reply]
Thanks!

**Reply by lindosgm** [reply]
Thanks, this worked.

**Reply by kacperstank** [reply]
Thanks a lot man! Made my day.

**Reply by masakudamatsu** [reply]
This worked for my iMac. Thanks a lot!

**Reply by EduardBel** [reply]
This worked for me too. Thanks!!

**Reply by tjmurray** [reply]
This is the best solution since it kills two birds with one stone: bypasses networking page and also bypasses the requirement to log in with a Microsoft account.

As posted above by chrisherman: if you cannot locate the backslash, you can do "cd oobe" and "bypassnro". Each followed by pressing enter-key

**Reply by harshanas** [reply]
This worked for me. Thank you

**Comment by Vaidas737** [comment]
This worked for me:

On the error screen press Fn+ Ctrl+ Shift + F3 and it will boot you into an admin profile. Then mount the drive with the splice tools and install it. Then reboot the machine and you will be connected to your mac's Internet and it will take you though the rest of the steps

**Reply by dhess** [reply]
Thank you!

Just a note to anyone who uses this method: simply rebooting will cause the VM to boot into Administrator mode again. To be clear, after performing this step, you'll be presented with a dialog box where you can set up the "out of box experience". Just press the "OK" button, let it do its thing, and then reboot when prompted by the dialog box. After reboot, you should be able to advance past the network check.

**Reply by richardschoen** [reply]
This works great. I am on UTM 4.0.8.

**Reply by tjmurray** [reply]
This works well, thank you. Though note that with this method you will still be forced to log in with a Microsoft account. The alternative "oobe\bypassnro" solution higher in the thread solves both problems so might be preferable for some users.

**Comment by EugeneSwagzit** [comment]
Important question.

After setting this up, perfect, I'm in, no problem. But having used the OOBE\BYPASSNRO command during the onboarding process, I am now into the main system with no means of activating my Internet connection. This same code wiped the option to toggle Wifi in the first place. Obviously, this kind of limits my options, so how would I go about restoring this function?

I'm hoping there's like another code or something simple I can use as opposed to having to reinstall some abstract version of Windows 11 on UTM, but I'll take what I can get.

**Reply by dna246** [reply]
It does not wipe the option Eugene, at least that's what I think. 
Close your virtual window, then click on  UTM settings (the setting icon on top right corner), 
then Network, then you'll see Network Mode change it to Bridged Network. 
It worked for me.

**Comment by tony1tf** [comment]
On my MacbookPro M2 laptop,  it all worked OK by installing the utilities
that UTM provide. Although there didn't seem to be the standard Windows
Internet app visible on the screen,  the WiFi from the Mac side seemed to
just be there!
I'm not running it presently though because the processor usage went up to
nearly 400%and was rapidly discharging the battery, so there might be
another issue.

**Comment by pytransformer** [comment]
I've followed all the steps until the shift + f10. It doesn't seem to work for me, and I've tried many times, but it just doesn't launch the Windows Command Line. I have a M1 MacBook Air.

**Reply by thomask** [reply]
First press `shift + F10` or (`fn + shift + F10`). Then press `cmd+r` and type `cmd` to open the command prompt

**Reply by cullylarson** [reply]
This worked for me. Thanks! One thing I had to do before running `oobe\bypassnro` was to `cd \Windows\System32`. So:

```
cd \Windows\System32
oobe\bypassnro
```

**Reply by maxd-out83** [reply]
When I do that, It says error, Access Denied.

**Comment by verma-rajatk** [comment]
![image](https://user-images.githubusercontent.com/19155094/214964129-0e689211-37d6-45b3-8e2e-7fc972c9c3fa.png)
These settings worked for me on Mac M1

**Reply by twoarmtom** [reply]
This solved it for me as well.

**Reply by antonio-bravo** [reply]
Thank you so much @verma-rajatk , choosing Network Mode = Emulated VLAN worked for me too.

**Reply by caugner** [reply]
Choosing Network Mode = Emulated VLAN didn't work for me just now, but [this approach](https://github.com/utmapp/UTM/discussions/3795#discussioncomment-3080091) worked.

**Comment by kisjoke91** [comment]
Most recently (22598) the OOBE thingy doest work out pretty well. 
Once you have to select the region, fire up the cmd, open explorer with it, and install spice tools

**Reply by Sibyx** [reply]
Yes! This worked like a dream!

**Comment by dixonge** [comment]
In VM settings I selected 'bridged advanced' and selected en1 (because my Mac Mini M1 is on WiFi only) - on reboot the VM had full internet access.
